### ISSUE TITLE
Typescript Error

### ISSUE THREAD

**Issue by osseonews** [commercelayer/commercelayer-cart#21]
In the file, components/cart/index.tsx the code below throws a TS error per the error below. How is this TS error removed?

 ```
<LineItemsCount>
                  {({ quantity }) =>
                    quantity ? (
                      <div className="text-sm text-gray-500 font-semibold">
                        <span data-test-id="items-count">{quantity}</span>{" "}
                        {t("general.item", { count: quantity })}
                      </div>
                    ) : null
                  }
                </LineItemsCount>
```

The error reported is:
```
Type '({ quantity }: Omit<LineItemsCountProps, "children"> & { quantity: number; }) => Element | null' is not assignable to type '(LineItemsCountChildrenProps & (string | number | boolean | ReactElement<any, string | JSXElementConstructor<any>> | ReactFragment | ReactPortal | null)) | undefined'.
  Type '({ quantity }: Omit<LineItemsCountProps, "children"> & { quantity: number; }) => Element | null' is not assignable to type 'LineItemsCountChildrenProps & string'.
    Type '({ quantity }: Omit<LineItemsCountProps, "children"> & { quantity: number; }) => Element | null' is not assignable to type 'string'.ts(2322)
```

**Comment by gciotola** [comment]
Hi @osseonews,
I was looking into this and unfortunately, I could not reproduce the issue. 
Where do you get the TS error? 
The build task (`yarn run build`) works fine and type-checking is enabled.
Are you able to build the app?

I remember probably seeing this when trying to use React 18, which is not supported yet (we are working on it).

**Comment by gciotola** [comment]
Hi @osseonews 
Sorry if this took a while. We have just released a new cart [version 2.0.0](https://github.com/commercelayer/commercelayer-cart/releases/tag/v2.0.0) which also includes the update to React 18 (using @commercelayer/react-components v4).
This solves the issue you had with Typescript when using react 18.

Please notes that this version comes with some breaking change since it does not use NextJs anymore, but a much simpler `Vite.js` build that can be deployed on any static site web hosting and does not require any server-side capabilities.
You can check the release notes to see what's changed.

Any feedback is appreciated.
